cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/MapReduceMetricsWriter.java

```diff
@@ -112,8 +133,25 @@ private void reportMapredStats(Counters jobCounters) throws IOException, Interru
               runningReducers * memoryPerReducer);
   }
 
+  private void reportMapTaskMetrics(TaskReport taskReport) {
+    Counters counters = taskReport.getTaskCounters();
+    MetricsCollector metricsCollector = mapTaskMetricsCollectors.getUnchecked(taskReport.getTaskId());
+    metricsCollector.gauge(METRIC_INPUT_RECORDS, getTaskCounter(counters, TaskCounter.MAP_INPUT_RECORDS));
+    metricsCollector.gauge(METRIC_OUTPUT_RECORDS, getTaskCounter(counters, TaskCounter.MAP_OUTPUT_RECORDS));
+    metricsCollector.gauge(METRIC_BYTES, getTaskCounter(counters, TaskCounter.MAP_OUTPUT_BYTES));
+    metricsCollector.gauge(METRIC_COMPLETION, (long) (taskReport.getProgress() * 100));
```

Comment:
question: when you query for metrics at job level , say "METRIC_COMPLETION", that will include these task level completion as well right ? ie (job-level completion + task-level-completion * num-tasks) while we just want job-level info, how are we tackling that ?